### PR TITLE
[4.0] Fix for AccessibleMediaField

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for Issue # https://github.com/joomla/joomla-cms/issues/29825.

### Summary of Changes
If the `accessiblemedia` field within in a `subfield`, `value` is an object `stdClass`. I extended the check for this. Also, I added the unit tests for the field and do not use `$field->apivalue`any more because the name is misleading and it is not possible in a `subfield` custom field.

### Testing Instructions

1. Please test like explained here: https://github.com/joomla/joomla-cms/pull/27712

2. Please also check whether the media custom field can be used within a subform custom field. Note: If you insert the media custom field in a subform custom field, you can only select (when editing the article) an image file after you have saved the article. This issue is not related to this PR. Please see https://github.com/joomla/joomla-cms/issues/26711.

3. Check also the rendering in the front end for the custom field of type `media` and `subfield incluing media field`.

### Documentation Changes Required
Yes

### CODE REVIEW

**Review by wilsonge — APPROVED** [review]

File: tests/Unit/Libraries/Cms/Form/Field/AccessiblemediaFieldTest.php [hunk 1]
```diff
@@ -190,6 +190,56 @@ public function testSetupWithValueThatIsNotValid()
...
+
+		$element = new \SimpleXMLElement('<field name="testfield" />');
+
+		$this->assertTrue($accessiblemediafield->setup($element, $obj, null));
+		$this->assertEquals($obj, $accessiblemediafield->__get('value'));
+	}
+
+	/**
+	 * Tests method to attach a Form object to the field.
+	 * If the value is a stdClass that is not valid for the accessible meia field.
```
Comment (Quy) [inline tests/Unit/Libraries/Cms/Form/Field/AccessiblemediaFieldTest.php]:
```suggestion
	 * If the value is a stdClass that is not valid for the accessible media field.
```

Comment (astridx) [reply tests/Unit/Libraries/Cms/Form/Field/AccessiblemediaFieldTest.php]: Thanks

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 62c45d297c57acbaabe0df214c8f7cb4fbcfcf9d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29840">issues.joomla.org/tracker/joomla-cms/29840</a>.</sub>

**Comment by wilsonge** [comment]
The API Value allows you to override the value returned by the API application. Nothing more.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**